src/api/store/supported_language.py

```diff
@@ -0,0 +1,77 @@
+from database.models import SupportedLanguage
+from _main_.utils.massenergize_errors import CustomMassenergizeError
+from _main_.utils.activity_logger import log
+from typing import Tuple
+from django.db import IntegrityError
+
+log = log.exception
+
+class SupportedLanguageStore:
+    def __init__ (self):
+        self.name = "Supported Language Store/DB"
+
+    def get_supported_language_info (self, args) -> Tuple[ dict or None, any ]:
+        try:
+            language_code = args.get('language_code', None)
+
+            if not language_code:
+                return None, CustomMassenergizeError("Please provide a valid id or code")
+
+            language = SupportedLanguage.objects.filter(code = language_code).first()
+
+            return language, None
+        except Exception as e:
+            log(e)
```

Comment:
stay consistent with the pattern we have.
log.exception.  It makes refactoring easier.